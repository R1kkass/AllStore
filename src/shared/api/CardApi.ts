import axios from 'axios'
import { domen } from '../const/apiConst'

export interface ICardApi{
    products: ICard[]
    count: number
}

export interface ICard {
    id?: number
    imgUrl: string
    size: string
    brand: string
    code: string
    manufacturer: string
    price: number
    name: string
    count?: number
    description?: string
    type?: string[],
    productsid?: number
    images?:FileList
    categories?: string
    image?: [{
        imgUrl: string
        id: number
    }]
}

export interface ICardData{
    data: ICardApi[]
}

export interface ICardDataOne{
    data: ICard
}

export const CardApi = async()=>{
    let res:ICardData = await axios.get(`${domen}/getall`) 
    return res
}

export const SeacrchApi = async(searchParam: any)=>{
    let res:ICardData = await axios.post(`${domen}/getall?search=${searchParam.get('search') || ""}&order=${searchParam.get('order') || "DESC"}&column=${searchParam.get('column') || "id"}&category=${searchParam.get('param') || ""}&page=${searchParam.get("page") || 1}`,
    {
        search: searchParam.get('search') || "",
        order: searchParam.get('order') || "DESC",
        column: searchParam.get('column') || "id",
        category: searchParam.get('param') || "",
        page: searchParam.get("page") || 1
    }) 
    console.log(res);
    
    return res
}

export const DeleteCardApi = async(id:number)=>{
    let res:ICardData = await axios.delete(`${domen}/admin/delete?id=`+id, {headers: {
        "Authorization": `Bearer ${localStorage.getItem("token")}`
    }})
    
    return res
}

export const AddCardApi = async(post:ICard) =>{
    const formdata = new FormData()
    formdata.append("code", post.code)
    formdata.append("name", post.name)
    formdata.append("brand", post.brand)
    formdata.append("count", String(post.count) || "1")
    formdata.append("description", post?.description || "")
    formdata.append("price", String(post?.price))
    formdata.append("size", String(post?.size))
    formdata.append("manufacturer", String(post.manufacturer))
    formdata.append("categories", String(post.categories))
    for(let i = 0; Number(post.images?.length) > i; i++){
        formdata.append("image[]", post?.images?.[i] || "")
    }
    
    let res:ICardData = await axios.post(`${domen}/admin/create`,formdata,
    {headers: {
        "Authorization": `Bearer ${localStorage.getItem("token")}`
    }}) 
    return res
}

export const EditCardApi = async(id: number, post: ICard)=>{
    let res:ICardData = await axios.post(`${domen}/admin/update?id=`+id,{
        ...post
    },
    {headers: {
        "Authorization": `Bearer ${localStorage.getItem("token")}`
    }})
    return res
}

export const CardApiId = async (id:string)=>{
    let res:ICardDataOne = await axios.post(`${domen}/getone?id=`+id, {id: id}) 
    return res.data
}