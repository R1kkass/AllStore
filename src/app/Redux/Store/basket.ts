import { IResponseDataBasket } from "../../../shared/api/BasketApi"
import { ICardApi } from "../../../shared/api/CardApi"

const initialState = {
    basket: [],
    count: JSON.parse(localStorage.getItem('basket') || '[]')?.length || 0,
}

interface IAction {
    type: string
    basket: IResponseDataBasket[]
    count: number
}

export const ADD_BASKET = "ADD_BASKET"
export const ADD_COUNT = "ADD_COUNT"

export const basketReducer = (state = initialState, action: IAction) => {
    switch (action.type) {
        case ADD_BASKET:
            
            return { ...state, basket: action.basket, count: action.basket.length }
        case ADD_COUNT:
            return { ...state, basket: action.count }
        default:
            return state
    }
}

export const addBasket = (posts:IResponseDataBasket[])=>({type: ADD_BASKET, basket: posts})
export const addCount = (count:number)=>({type: ADD_COUNT, count: count})
