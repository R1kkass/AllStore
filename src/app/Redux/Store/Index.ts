import { combineReducers, createStore } from "redux"
import { ICard, ICardApi } from "../../../shared/api/CardApi"
import { basketReducer } from "./basket"
import { productReducer } from "./product"
import errorButtonReducer from "./errorButton"
import orderReducer from "./orders"
import loaderReducer from "./loader"
import { composeWithDevTools } from 'redux-devtools-extension'
import newsReducer from "./newsPanel"

export interface IRedux {
    basket: ICardApi
    product: { posts: ICard[], count: number }
}

const rootReducers = combineReducers({
    basket: basketReducer,
    product: productReducer,
    errorButton: errorButtonReducer,
    order: orderReducer,
    loader: loaderReducer,
    news: newsReducer
})

export const store = createStore(rootReducers, composeWithDevTools())
