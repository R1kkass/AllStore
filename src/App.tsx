import Routing from "./pages"
import "./app/Styles/GlobalStyles.scss"
import { Provider } from "react-redux"
import { store } from "./app/Redux/Store/Index"

function App() {

    return (
        <Provider store={store}>
                <Routing />
        </Provider>
    )
}

export default App
