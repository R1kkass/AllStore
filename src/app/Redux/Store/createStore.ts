import { store } from "./Index";

export type RootState = ReturnType<typeof store.getState>;