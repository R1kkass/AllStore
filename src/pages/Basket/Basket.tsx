import { Fragment, useContext, useEffect, useState } from "react"
import { useDispatch, useSelector } from "react-redux"
import { addBasket } from "../../app/Redux/Store/basket"
import Breadcrumbs from "../../features/BreadCrumbs/BreadCrumbs"
import { ICardApi } from "../../shared/api/CardApi"
import MyButton from "../../shared/UI/Buttons/MyButton/MyButton"
import CardLine from "../../entities/Card/CardLine"
import Modal from "../../shared/UI/Modal/Modal"
import "./Basket.scss"
import { GetBasket, IResponseDataBasket } from "../../shared/api/BasketApi"
import { Link } from "react-router-dom"
import Counter from "../../features/Counter/Counter"
import BasketList from "../../widgets/BasketList/BasketList"


const Basket = () => {
    const dispatch = useDispatch()
    const count: number = useSelector((state: any) => state.basket.count)
    document.title = "Корзина"

    useEffect(() => {
        GetBasket().then(e=>
            dispatch(addBasket(e))            
        )
    }, [count])

    return (
        <div className="Basket">
            <Breadcrumbs
                arr={[
                    { name: "Главная", link: "/" },
                    { name: "Каталог", link: "/catalog" },
                    { name: "Корзина", link: "/basket" },
                ]}
            />
            <h1>Корзина</h1>
            <BasketList/>
            {count ? (
                <div className="Basket__order">
                    <div>
                        <Link to="/order">
                            <MyButton>
                                Оформить заказ
                            </MyButton>
                        </Link>
                    </div>
                    <div>
                        <Counter/>
                    </div>
                </div>
            ) : (
                <div className="Basket__order">
                    <h1>Корзина пуста</h1>
                </div>
            )}
        </div>
    )
}

export default Basket