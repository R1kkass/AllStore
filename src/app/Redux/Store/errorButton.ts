import { RootState } from "./createStore";
import { PayloadAction, createSlice } from "@reduxjs/toolkit";
export const LOAD_FILM = "LOAD_FILM";

interface IInit {
    errorButton: IVisib[]
}

const initialState: IInit = {
    errorButton: [{
        id: 1,
        visible: false,
        text: "Ошибка",
        type: "Error"
    }]
};

export interface IVisib{
    id: number;
    visible: boolean,
    text: string,
    type?: "Error" | "Success"
}

const filmSlice = createSlice({
    name: "errorButton",
    initialState,
    reducers: {
        errorButtonAction(state: IInit, action: PayloadAction<IVisib>) {
            state.errorButton = [...state.errorButton, action.payload] 
        },
        deleteErrorAction(state: IInit, action: PayloadAction<number>) {
            let newState = state.errorButton.filter((err)=>{
                return err.id!==action.payload
            })
            state.errorButton = [...newState]
        }
    }
});

const { reducer: errorButtonReducer, actions } = filmSlice;

export const getAllError =
    () =>
        (state: RootState): IVisib[] =>
            state.errorButton.errorButton;

export const addError = () => {

}
export const { errorButtonAction, deleteErrorAction } = actions;
export default errorButtonReducer;