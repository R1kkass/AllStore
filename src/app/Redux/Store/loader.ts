import { RootState } from "./createStore";
import { PayloadAction, createSlice } from "@reduxjs/toolkit";
export const LOAD_FILM = "LOAD_FILM";

interface IInit {
    loader: boolean
}

const initialState: IInit = {
    loader: false
};

const loaderSlice = createSlice({
    name: "loader",
    initialState,
    reducers: {
        loaderAction(state: IInit, action: PayloadAction<boolean>) {
            state.loader = action.payload;
        }
    }
});

const { reducer: loaderReducer, actions } = loaderSlice;
export const getLoader =
    () =>
        (state: RootState): boolean =>
            state.loader.loader;

export const { loaderAction } = actions;
export default loaderReducer;