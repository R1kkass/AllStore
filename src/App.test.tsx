import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';
import Basket from './pages/Basket/Basket';
import { Provider, useSelector } from 'react-redux';
import {store} from '../src/app/Redux/Store/Index'
import { MemoryRouter } from 'react-router-dom';
import { addBasket } from './app/Redux/Store/basket';
import { ICardApi } from './shared/api/CardApi';

const res:ICardApi[] = [{
  id: 1,
  imgURL: '',
  size: '',
  brand: '',
  code: '',
  manufacturer: '43',
  price: 432,
  name: '',
  count:1,
  description:'',
  type: ['fd','fd']
}]

  test('Empty arr', () => {
    let tree = render(<MemoryRouter><Provider store={store}><Basket /></Provider></MemoryRouter>);
    let res = screen.getByText(/Корзина пуста/i)
    expect(res).toBeInTheDocument();
  });

  beforeEach(()=>{
    addBasket([{
        id: 1,
        imgURL: '',
        size: '',
        brand: '',
        code: '',
        manufacturer: '43',
        price: 432,
        name: '',
        count:1,
        description:'',
        type: ['fd','fd']
      }])
  })

  test('With arr', () => {
    // 
    let tree = render(<MemoryRouter><Provider store={store}><Basket /></Provider></MemoryRouter>);

    expect(tree).toMatchSnapshot();
  });