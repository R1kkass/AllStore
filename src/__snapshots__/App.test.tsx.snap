// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With arr 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Basket"
      >
        <div
          class="BreadCrumbs"
        >
          <a
            href="/"
          >
            Каталог
             
            <img
              src="Crumb.svg"
            />
             
          </a>
          <a
            href="/basket"
          >
            Корзина
             
            <img
              src="Crumb.svg"
            />
             
          </a>
        </div>
        <h1>
          Корзина
        </h1>
        <div
          class="Basket__order"
        >
          <h1>
            Корзина пуста
          </h1>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Basket"
    >
      <div
        class="BreadCrumbs"
      >
        <a
          href="/"
        >
          Каталог
           
          <img
            src="Crumb.svg"
          />
           
        </a>
        <a
          href="/basket"
        >
          Корзина
           
          <img
            src="Crumb.svg"
          />
           
        </a>
      </div>
      <h1>
        Корзина
      </h1>
      <div
        class="Basket__order"
      >
        <h1>
          Корзина пуста
        </h1>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
